### PR TITLE
fixed category

### PR DESCRIPTION
**Pull request by mvn-tienho-dn**
create, list,edit,delete, and config path for category

### CODE REVIEW

File: app/Http/Backend/Controllers/CategoryController.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+class CategoryController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $categories = Category::select('id', 'name', 'image')->orderBy('id', 'DESC')->paginate(2);
```
Comment (ghost) [inline app/Http/Backend/Controllers/CategoryController.php]: why paginate 2

File: app/Http/Backend/Controllers/CategoryController.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('backend.categories.add');
```
Comment (ghost) [inline app/Http/Backend/Controllers/CategoryController.php]: view name should be `create`

**Comment by ghost** [comment]
@thienvuong Please review this PR

File: app/Http/Backend/Controllers/CategoryController.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+     * @param \Illuminate\Http\Request $request request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(AddCategoryRequest $request)
+    {
+        $img = $request->file('image');
+        $imgName = time().'-'.$img->getClientOriginalName();
+        $categories = new Category($request->all());
+        $categories->name = $request->name;
```
Comment (at-huydoan) [inline app/Http/Backend/Controllers/CategoryController.php]: redundant, already in $request->all()

File: app/Http/Backend/Controllers/CategoryController.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        $categories->image =$imgName;
+        $categories->save();
+        $des = Category::IMAGES_PATH;
+        $img->move($des, $imgName);
```
Comment (at-huydoan) [inline app/Http/Backend/Controllers/CategoryController.php]: save after move . write function upload image is better

File: app/Http/Backend/Controllers/CategoryController.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+     *
+     * @param int $id id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $categories = Category::find($id);
+        return $categories;
+    }
```
Comment (at-huydoan) [inline app/Http/Backend/Controllers/CategoryController.php:64]: return view.

File: app/Http/Backend/Controllers/CategoryController.php [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            $category->name  = $request->name;
+            $category->image = $imgName;
+            $category->save();
+            $des = Category::IMAGES_PATH;
+            $img->move($des, $imgName);
+            return redirect()->route('admin.categories.index');
+        } else {
+            $category = Category::find($id);
+            $category->name = $request->name;
+            $category->save();
```
Comment (at-huydoan) [inline app/Http/Backend/Controllers/CategoryController.php]: you can check before save, write function upload in models.

File: app/Http/Backend/Requests/AddCategoryRequest.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+     **/
+    public function messages()
+    {
+        return [
+            'name.required'     =>'Vui lòng nhập tên chuyên mục !',
+            'image.image'       =>'Bạn phải nhập file hình ảnh !',
+            'image.max'         =>'Dung lượng bức ảnh không được quá 2 Mb !',
+            'image.required'    =>'Vui lòng chọn hình ảnh!',
+        ];
+    }
```
Comment (at-huydoan) [inline app/Http/Backend/Requests/AddCategoryRequest.php]: use default of laravel.

File: app/Http/Backend/Requests/EditCategoryRequest.php [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function messages()
+    {
+        return[
+            'name.required'     =>'Vui lòng nhập tên chuyên mục !',
+            'image.image'       =>'Bạn phải nhập file hình ảnh !',
+            'image.max'         =>'Dung lượng bức ảnh không được quá 2 Mb !',
+        ];
+    }
```
Comment (at-huydoan) [inline app/Http/Backend/Requests/EditCategoryRequest.php]: use default of laravel.

File: app/Http/Backend/Controllers/CategoryController.php [hunk 9]
```diff
@@ -18,7 +18,7 @@ class CategoryController extends Controller
      */
     public function index()
     {
-        $categories = Category::select('id', 'name', 'image')->orderBy('id', 'DESC')->paginate(2);
+        $categories = Category::select('id', 'name', 'image')->orderBy('id', 'DESC')->paginate(10);
```
Comment (at-huydoan) [inline app/Http/Backend/Controllers/CategoryController.php:21]: $perPage = 10 in models

File: app/Http/Backend/Controllers/CategoryController.php [hunk 10]
```diff
@@ -88,22 +86,14 @@ public function edit($id)
      */
     public function update(EditCategoryRequest $request, $id)
     {
+        $category = Category::find($id);
```
Comment (at-huydoan) [inline app/Http/Backend/Controllers/CategoryController.php]: use findOrFail

File: resources/views/backend/categories/create.blade.php [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+      </ol>
+   </section>
+
+   <!-- Main content -->
+   <section class="content">
+
+   <div class="row">
+        <div class="col-md-6">
+           <h2 style="color:red;">Insert Category</h2>
+                   {!!Form::open([
```
Comment (at-huydoan) [inline resources/views/backend/categories/create.blade.php]: use {{ for form

**Comment by at-huydoan** [comment]
@at-vienlm please review this PR

**Comment by ghost** [comment]
please fix conflict. thanks

**Pull request closed without merging**